scripts/error-docgen.mjs

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable no-mixed-spaces-and-tabs */
+import fs from 'fs';
+import jsdoc from 'jsdoc-api';
+import fetch from 'node-fetch';
+import ts from 'typescript';
+
+import '../../astro/packages/astro/dist/core/errors/errors-data.js';
+
+const errorURL =
+	'https://raw.githubusercontent.com/withastro/astro/main/packages/astro/src/core/errors/errors-data.ts';
+
+// Fill this in to test a response locally, with fetching.
+const STUB = undefined; // fs.readFileSync('../astro/packages/astro/src/core/errors/errors-data.ts', {encoding: 'utf-8',});
+
+const compilerErrorURL =
+	'https://raw.githubusercontent.com/withastro/compiler/main/packages/compiler/shared/diagnostics.ts';
+
+// Fill this in to test a response locally, with fetching.
+const compilerSTUB = undefined; // fs.readFileSync('../compiler/packages/compiler/shared/diagnostics.ts', {encoding: 'utf-8',});
+
+const HEADER = `---
+# NOTE: This file is auto-generated from 'scripts/error-docgen.mjs'
+# Do not make edits to it directly, they will be overwritten.
+# Instead, change this file: https://github.com/withastro/astro/blob/main/packages/astro/src/core/errors/errors-data.ts
+# Translators, please remove this note and the <DontEditWarning/> component.
+
+layout: ~/layouts/MainLayout.astro
+title: Error reference
+i18nReady: true
+githubURL: https://github.com/withastro/astro/blob/main/packages/astro/src/core/errors/errors-data.ts
+setup: |
+  import Since from '../../../components/Since.astro';
+  import DontEditWarning from '../../../components/DontEditWarning.astro';
+---
+
+<DontEditWarning />
+
+The following reference covers all errors that can be emitted by Astro. To learn more about common pitfalls, see our guide on [Troubleshooting](/en/guides/troubleshooting/).
+
+`;
+
+const FOOTER = ``;
+
+export async function run() {
+	const compilerErrorData = { errors: [], jsdoc: [] }; // await getCompilerErrors();
```

Comment:
Compiler errors are unintentionally not covered in this PR. I'll tackle it separately since it potentially requires some work in the error shapes from the compiler